lib/active_fedora/rdf/identifiable.rb

```diff
@@ -47,7 +47,7 @@ def resource_datastream(ds=nil)
     # @param [RDF::URI] uri URI that is being looked up.
     def from_uri(uri,_)
       begin
-        self.find(pid_from_subject(uri))
+        ActiveFedora::Base.find(pid_from_subject(uri))
```

Comment:
@jcoyne Because it blows up now if the uri points to an AF object of a different model (i.e., not <= self.class).  Alternatively, I suppose we could use `cast: false` but this seems better to me.